### ISSUE TITLE
unet.eval() not giving the same results as unet.train()

### ISSUE THREAD

**Issue by aniqueakhtar** [facebookresearch/SparseConvNet#166]
I am facing this weird issue that wasn't there before. 
When I run ``unet.eval()`` the output of the network is totally different from what it should be in ``unet.train()``

When I do unet.train(), I get these results:
![Screenshot from 2020-04-07 12-18-47](https://user-images.githubusercontent.com/2750083/78699852-2b18e480-78ca-11ea-9341-f7cd27272020.png)

When I do unet.eval(), I get these results
![Screenshot from 2020-04-07 12-19-13](https://user-images.githubusercontent.com/2750083/78699862-31a75c00-78ca-11ea-819b-8d62c4a87543.png)

Please do suggest what might be causing this? 
The network is not changing and they are both being fed the same data.
Thank you

**Comment by btgraham** [comment]
In eval mode, the batchnorm is using saved batch statistics, which can results in larger values in the output. What does the input look like?

**Comment by aniqueakhtar** [comment]
Yeah, I figured the batchnorm was the issue here.
The input is 6 batches of .long() with scale = 2**4 and full_scale = 2**10, with the number of points ranging from 2k to 10k.
The interesting thing is I can get the correct results by editing these settings:
1) If I use scale = 1 and fulle_scale = 2**6.
2) If I keep a fixed number of points in each input i.e. 12k with batchsize of 4, and use scale = 2**4 and full_scale = 2**10 values.

I am unsure why the above two methods seem to be working but I would still like my original method to work cause I think it is the most optimum solution for my problem.

I am using my output as residual output for denoising so it is important that the output is the same during training and eval. Is there a way I can fix this and use it for a higher scale and full_scale value?
